src/gui/modals/ControllerConnectionDialog.cpp

```diff
@@ -373,7 +372,11 @@ void ControllerConnectionDialog::midiToggled()
 			connect( m_midiController, SIGNAL(valueChanged()), this, SLOT(midiValueChanged()));
 		}
 	}
-	m_midiAutoDetect.setValue( enabled );
+
+	// TODO: This should be reworked into some sort of "shadow" state rather than turning it off...
+	// since it makes for switching between the USER mode and CONTROLLER mode a bit awkward
+	// If you toggle the autodetect then move from MIDI to USER then back to MIDI it resets, which is not ideal.
+	m_midiAutoDetect.setValue( enabled && (!m_targetModel->controllerConnection()) );
```

Comment:
```suggestion
	m_midiAutoDetect.setValue(enabled && (!m_targetModel->controllerConnection()));
```

Usual white space issue